### ISSUE TITLE
Use TF trained model

### ISSUE THREAD

**Issue by fedegarza** [jolibrain/deepdetect#273]
Hi, I'm trying to use this trained model (inception v3)
https://storage.googleapis.com/download.tensorflow.org/models/inception_dec_2015.zip
from TF samples https://www.tensorflow.org/versions/master/tutorials/image_recognition/

but it doesn't work. When I try to create a service I get
stdClass Object
(
    [status] => stdClass Object
        (
            [code] => 500
            [msg] => InternalError
            [dd_code] => 1007
            [dd_msg] => stoi
        )

)

Can I use TF pre-trained models directly en DD? 
I have a TF pre-trained model previously that I've installed DD and I would like to use it

Thanks!

**Comment by beniz** [comment]
Have you looked at the list of available models and checked that they work with your setup ? Also, please use the issue template, specify your OS, setup, provide all API calls, etc...

**Comment by fedegarza** [comment]
Sorry about the template issue
I'll do my best

#### Configuration
- Version of DeepDetect: 
    - [x] Locally compiled on:
        - [x] Ubuntu 14.04 LTS
          - Bazel 0.4.3
          - Protobuf 3.1.0
          
- Commit (shown by the server when starting):
  DeepDetect [ commit 0d8ca6f3dbcda31d48d01eea453316ebcbbfad1a ]

#### Your question / the problem you're facing:
I'm trying to use a TF pre-training model but it doesn't work.
https://storage.googleapis.com/download.tensorflow.org/models/inception_dec_2015.zip

I have tried with other model published in your web and they worked right.  

#### Error message (if any) / steps to reproduce the problem:
- [ ] list of API calls:
     curl -X PUT "http://localhost:8080/services/myserv" -d '{
                "mllib":"tensorflow",
                "description":"Inception Dec 2015 Image Classification Service",
                "type":"supervised",
                "parameters":{
                    "input":{
                        "connector":"image",
                        "height":299,
                        "width":299
                    },
                    "mllib":{
                        "nclasses":1000,
                        "inputlayer":"InputImage"
                    }
                },
                "model":{
                    "repository":"/opt/deepdetect/models/base/tf/inception_dec_2015"
                }
            }'

- [ ] Server log output:
    
    API response
    stdClass Object
    (
        [status] => stdClass Object
            (
                [code] => 500
                [msg] => InternalError
                [dd_code] => 1007
                [dd_msg] => stoi
            )
    
    )
    
    SERVER RESPONSE
    E /opt/deepdetect/src/httpjsonapi.cc:392] Mon Mar 13 16:44:05 2017 UTC - xx.xx.xx.233 "POST /services/myserv" 500 0


Thanks!

**Comment by beniz** [comment]
Thanks, can you list the files in `/opt/deepdetect/models/base/tf/inception_dec_2015` ?

**Comment by fedegarza** [comment]
There are 2 files
imagenet_comp_graph_label_strings.txt  
tensorflow_inception_graph.pb

thanks!

**Comment by beniz** [comment]
have you tried using one of the TF models we provide in the README ?

**Comment by fedegarza** [comment]
Yes, I tried with 
https://deepdetect.com/models/tf/inception_v3.pb -- works
https://deepdetect.com/models/tf/openimages_inception_v3/ -- works

and over https://deepdetect.com/applications/model/#lic
Clothing and Footwear -- work right both

If I train a model on TF in another machine, then can I use it on DD? 
Thaks!

**Comment by beniz** [comment]
You can but .pb file must be produced out of the .cpk file you obtain from training with TF. It is not a DD thing, but TF C++ API that to the best of my knowledge requires the protobuf pb file.

**Comment by beniz** [comment]
We actually have a script for the `ckpt` to `pb` file conversion that we haven't released because it is very much adhoc. It is based on the freeze_graph code from TF. I'll put a gist out there with the script if it is needed.

**Comment by fedegarza** [comment]
ok, this sounds very interesting. Thank you for your help

I'll take a look in how TF are saving the model, but it should work directly.

Do you know how can I debug a little more TF or ml side on API calls in DD? 

Thanks again

**Comment by beniz** [comment]
> Do you know how can I debug a little more TF or ml side on API calls in DD?

Make sure that your file `imagenet_comp_graph_label_strings.txt` has the correct format, you can look at the `corresp.txt` file in other models, such as https://deepdetect.com/models/tf/corresp_inception_clean.txt

**Comment by fedegarza** [comment]
Hi, thanks for your response
It works, it seems the txt hasn't got the first column. Actually, I don't know it was necessary. I tried one model without the txt and it worked right, but in prediction, it didn't returned the label (openimage_inception_v3)

If I change or remove the txt, I have no more the last error and I can create a service, but now I get the next error

E /opt/deepdetect/src/services.h:490] service inception_v3_dec_2015 mllib internal error: Not found: FeedInputs: unable to find feed output InputImage

Any idea???
Thanks a lot!

**Comment by beniz** [comment]
look the API up for inputlayer and outputlayer.

**Comment by fedegarza** [comment]
Thanks for your help 

If I set the inputlayer parameter [InputImage] I get an error 500 FeedInputs: unable to find feed output InputImage
I read the API documentation and search about this parameter but I couldn't found anything about it, so I tried removing it.
Without parameter, I can create the service but then at prediction, I get another 500 error about Internal: Output 0 of type float does not match declared output type string for node _recv_DecodeJpeg/contents_0

I don't know if it's related to the model ( .pb from 2015) or to the API.

I'll try to train another version and we'll see.

Thanks again!
